File: first_duplicate_letter.rb
# Given a string, write a function that returns the first occurence of two duplicate characters in a row, and return the duplicated character.

# Input: “abcdefghhijkkloooop”
# Output: “h”

def first_duplicate_letter(string)
  count = {}
  count.default = 0
  index = 0
  downcase_string = string.downcase
  while index < downcase_string.length
    count[downcase_string[index]] += 1
    if count[downcase_string[index]] == 2
      output = downcase_string[index]
      break
    end
    index += 1
  end
  return output
end

p first_duplicate_letter("abcdefghhijkkloooop")

File: c++/first_duplicate_letter.cpp
// Given a string, write a function that returns the first occurence of two duplicate characters in a row, and return the duplicated character.

// Input: “abcdefghhijkkloooop”
// Output: “h”

#include <iostream>
#include <string>
#include <map>
using namespace std;

char duplicate(string input);

int main()
{
  cout << duplicate("abcdefghhijkkloooop") << endl;
  return 0;
}

char duplicate(string input)
{
  char output;
  map<char, int> count;

  for (int i = 0; i < input.length(); i++)
  {
    count[input[i]]++;
    if (count[input[i]] > 1)
    {
      output = input[i];
      break;
    }
  }

  return output;
}